src/Compilers/CSharp/Test/Syntax/Parsing/LambdaAttributeParsingTests.cs

```diff
@@ -3637,5 +3641,868 @@ void verify(string source, ParseOptions? parseOptions = null)
+                                                                            N(SyntaxKind.CloseParenToken);
+                                                                        }
+                                                                        N(SyntaxKind.EqualsGreaterThanToken);
+                                                                        N(SyntaxKind.Block);
+                                                                        {
+                                                                            N(SyntaxKind.OpenBraceToken);
+                                                                            N(SyntaxKind.CloseBraceToken);
+                                                                        }
+                                                                    }
+                                                                }
+                                                                N(SyntaxKind.CloseParenToken);
+                                                            }
+                                                        }
+                                                    }
+                                                    N(SyntaxKind.CloseBracketToken);
+                                                }
+                                                N(SyntaxKind.ArgumentList);
+                                                {
+                                                    N(SyntaxKind.OpenParenToken);
+                                                    N(SyntaxKind.CloseParenToken);
+                                                }
+                                            }
+                                        }
+                                    }
+                                }
+                                M(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.Block);
+                            {
+                                N(SyntaxKind.OpenBraceToken);
+                                N(SyntaxKind.CloseBraceToken);
+                            }
+                            N(SyntaxKind.EmptyStatement);
+                            {
+                                N(SyntaxKind.SemicolonToken);
+                            }
+                            N(SyntaxKind.CloseBraceToken);
+                        }
+                    }
+                    N(SyntaxKind.CloseBraceToken);
+                }
+                N(SyntaxKind.EndOfFileToken);
+            }
+            EOF();
+        }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/73789")]
+        public void ParseAttributeWithLambda3_B()
+        {
+            UsingTree("""
+                class C
+                {
+                    void M()
+                    {
+                        // This is now an attributed lambda *within* a collection expression.  This is fine.
```

Comment:
>This is now an attributed lambda *within* a collection expression.  This is fine. [](http://example.com/codeflow?start=27&length=81)

The comment seems confusing since it sounds like the interpretation has changed. But this is the current and expected behavior.

Reply:
done.